File: src/main/java/com/wallet/app/service/PaymentService.java
package com.wallet.app.service;

import com.wallet.app.dto.CreatePaymentOrderRequest;
import com.wallet.app.dto.CreatePaymentOrderResponse;
import com.wallet.app.dto.VerifyPaymentRequest;
import com.wallet.app.dto.VerifyPaymentResponse;

public interface PaymentService {

    CreatePaymentOrderResponse createOrder(String username, CreatePaymentOrderRequest request);

    VerifyPaymentResponse verifyPayment(String username, VerifyPaymentRequest request);
}

File: frontend/src/services/payment.service.ts
import type {
  CreatePaymentOrderRequest,
  CreatePaymentOrderResponse,
  VerifyPaymentRequest,
  VerifyPaymentResponse,
} from '../types/api'
import { apiClient } from './apiClient'

export const paymentService = {
  async createOrder(payload: CreatePaymentOrderRequest): Promise<CreatePaymentOrderResponse> {
    const response = await apiClient.post<CreatePaymentOrderResponse>('/payments/create-order', payload)
    return response.data
  },

  async verifyPayment(payload: VerifyPaymentRequest): Promise<VerifyPaymentResponse> {
    const response = await apiClient.post<VerifyPaymentResponse>('/payments/verify', payload)
    return response.data
  },
}
